### PR TITLE
Fix relaunch command

### LINKED ISSUE #244: Refactor relaunch command
Current implementation is using `/launch/<launch_id>` endpoint to fetch previous workflow launch details. Better to use the endpoint `/workflow/<workflow_id>/launch` and mimic same Tower website UI behavior.

### PR DESCRIPTION
**Pull request by jordeu**
- Uses the endpoint `/workflow/<workflow_id>/launch` to get the launch to relaunch instead of `/launch/<launch_id>`
- Allows to set a run name when relaunching with `tw runs relaunch --name ...`
- Fixes #244

### CODE REVIEW

**Comment by jordeu** [comment]
@abhi18av @drpatelh @gwright99 can you help me, before doing a new release, check that this fixes the relaunch/resume?

To get the binaries of this branch you should go to the "Artifacts" sections (bottom right) of this page https://github.com/seqeralabs/tower-cli/actions/runs/2884122728

**Comment by gwright99** [comment]
Downloaded the artifact and tried to relaunch. Took me a few tries (see screenshot 1).

I assume that error about the session_id already being in use is expected? Got the same behaviour when I tried relaunching from the console (I tried relaunching the source a 2nd time while the 1st relaunched instance was still in flight).

image
image

**Comment by jordeu** [comment]
Thanks @gwright99! Those the 'Missing launch revision ID' error also happens from the web?

**Comment by gwright99** [comment]
> Thanks @gwright99! Does the 'Missing launch revision ID' error also happens from the web?

Nope. When I resume from the UI, the revision shows up as `master`:

#### Original Run
image

#### Resume from UI
image

#### Resumed Run
image

**Comment by jordeu** [comment]
I've fixed the discrepancy when relaunching a cancelled run that was not possible to resume. Now is doing like the website UI and it is relaunching it without trying to resume it. Still if it's possible to resume it, it will resume.

**Review by abhi18av — APPROVED** [review]
@jordeu , this is looking good to me ✅

File: src/test/resources/runcmd/runs/workflow_launch.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      "id": "2lu3NFms1qRvwTVnrs1yhg",
+      "name": "aws_batch",
+      "description": null,
+      "platform": "aws-batch",
+      "config": {
+        "region": "eu-west-1",
+        "computeQueue": "TowerForge-2lu3NFms1qRvwTVnrs1yhg-work",
+        "dragenQueue": null,
+        "computeJobRole": null,
+        "executionRole": "arn:aws:iam::128997144437:role/TowerForge-2lu3NFms1qRvwTVnrs1yhg-ExecutionRole",
```
Comment (abhi18av) [inline src/test/resources/runcmd/runs/workflow_launch.json:14]: Just raising a question here whether this file is necessary in the PR?

As I understand the automated cleans that Adianny has been implementing might kill this env/bucket anyhow no?

Comment (jordeu) [reply src/test/resources/runcmd/runs/workflow_launch.json:14]: This is just to mock the request/response of Tower API. So there is no interaction with AWS or Tower at all when running the unit tests.

Comment (abhi18av) [reply src/test/resources/runcmd/runs/workflow_launch.json:14]: Ah, okay - thanks Jordi.

**Pull request merged**